server/controllers/notifications.controller.js

```diff
@@ -21,16 +23,48 @@ function sendPushNotifications(req, res, next) { // eslint-disable-line no-unuse
             },
         },
     }).then((users) => {
-        if (!users) return;
+        if (!users.length > 0) {
+          res.status(404);
+          res.send({error: "Users Not Found"});
+          return
+        }
         users.forEach((user) => {
             if (!user) return;
-            const userData = pushData.find(_userData => _userData.username === user.username);
-            pushNotificationHelper.sendPushNotification(user, userData);
+            const userData = data.find(_userData => _userData.username === user.username);
+            switch (protocol) {
+            case 'push':
+                pushNotificationHelper.sendPushNotification(user, userData);
+                break;
+            case 'email':
+                emailHelper.sendEmail(user, userData);
+                break;
+            case 'sms':
+                smsHelper.sendSms(user, userData);
+                break;
+            default:
+            }
         });
         res.send({ success: true });
+    }).catch((err) => {
+      res.send({error: err})
     });
 }
 
+function sendPushNotifications(req, res, next) { // eslint-disable-line no-unused-vars
+    req.body.data = req.body.pushData;
```

Comment:
Because the API is being modified to support additional notification protocols. We can remove that line after all other microservices that currently use push are updated to send the payloads in the `data` attribute of the body.